src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -264,6 +271,8 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+// We made sure that browser never uses FileSystemWatcher.
+#pragma warning disable CA1416 // Validate platform compatibility
```

Comment:
Can you just put

```
[UnsupportedOSPlatform("browser")]
```

on these methods? there should be no reason to suppress this warning. the analyzer will see that you won't get here when on "browser".

Reply:
The only method it might not work for is `TryEnableFileSystemWatcher`

Reply:
There is one method (`TryEnableFileSystemWatcher`) that is called in both FSW and non-FSW code. I can annotate all these methods with `UnsupportedOSPlatform` but disable the warning in that specific call site.
https://github.com/dotnet/runtime/blob/47e82c1f625428c02eb6b31d7a7400c53a8c24b9/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs#L142-L145

Reply:
that sounds like a good approach to me